enginetest/histogram_test.go

```diff
@@ -0,0 +1,514 @@
+package enginetest
+
+import (
+	"container/heap"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+
+	"github.com/dolthub/go-mysql-server/memory"
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/rowexec"
+	"github.com/dolthub/go-mysql-server/sql/stats"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+func init() {
+	rand.Seed(0)
+}
+
+type statsTest struct {
+	name       string
+	tableGen   func(*sql.Context, *memory.Database, int, sql.TableId, sql.ColumnId, ...interface{}) *plan.ResolvedTable
+	args1      []interface{}
+	args2      []interface{}
+	leftOrd    []int
+	rightOrd   []int
+	leftTypes  []sql.Type
+	rightTypes []sql.Type
+	err        float64
+}
+
+func TestNormDist(t *testing.T) {
+	tests := []struct {
+		name  string
+		mean1 float64
+		std1  float64
+		mean2 float64
+		std2  float64
+	}{
+		{
+			name:  "same table",
+			mean1: 0,
+			std1:  10,
+			mean2: 0,
+			std2:  10,
+		},
+		{
+			name:  "same mean, different std",
+			mean1: 0,
+			std1:  10,
+			mean2: 0,
+			std2:  2,
+		},
+		{
+			name:  "peaks don't overlap",
+			mean1: 10,
+			std1:  5,
+			mean2: -10,
+			std2:  5,
+		},
+		//
+		{
+			name:  "same mean, different std1",
```

Comment:
The name suggests this test uses the same mean for both tables, but it doesn't. Some other tests here also have this confusion.